streams/src/main/java/org/apache/kafka/streams/kstream/Branched.java

```diff
@@ -143,6 +143,7 @@ protected Branched(final Branched<K, V> branched) {
      */
     @Override
     public Branched<K, V> withName(final String name) {
+        Objects.requireNonNull(name, "name cannot be null");
         return new Branched<>(name, chainFunction, chainConsumer);
```

Comment:
We have a similar method: `as` that accept a `name` parameter to create a Branched instance [here](https://github.com/apache/kafka/blob/trunk/streams/src/main/java/org/apache/kafka/streams/kstream/Branched.java#L53). We did a `name` null check there. We should also have a null check here for consistency.
```java
public static <K, V> Branched<K, V> as(final String name) {
        Objects.requireNonNull(name, "name cannot be null");
        return new Branched<>(name, null, null);
    }
```